definitions/npm/styled-components_v4.x.x/flow_v0.104.x-/test_styled-components_v4.x.x.js

```diff
@@ -117,29 +143,65 @@ describe('styled builtins', () => {
   })
 
   it('should inject theme', () => {
-    const Span: StyledComponent<{ color?: string, ... }, { accent: string, ... }, *> = styled.span`
+    const Span: StyledComponent<
+      { color?: string, ... },
+      { accent: string, ... },
+      *
+    > = styled.span`
       color: ${props => props.color || props.theme.accent};
     `
   })
 
   it('should validate theme', () => {
     // $ExpectError - oops, someone meant accent, not primary
-    const Span: StyledComponent<{ color?: string, ... }, { accent: string, ... }, *> = styled.span`
+    const Span: StyledComponent<
+      { color?: string, ... },
+      { accent: string, ... },
+      *
+    > = styled.span`
       color: ${props => props.color || props.theme.primary};
     `
   })
 
+  it('should support .attrs', () => {
+    const AttrsInput = styled.input.attrs<_, {| size: string |}>(
+      (props: any) => ({
+        // we can define static props
+        type: 'password',
+        // or we can define dynamic ones
+        margin: props => props.size,
```

Comment:
Is this correct? There is no need to create a function to access props, we already have access to `props`, also I think you meant `props.theme.size` here, but won't this type check with whatever you give it, because of the `props: any`?

Reply:
It should be just object, not function, I didn't mean `theme`

Reply:
I specified `any` because overload wasn't selecting function without annotation, which is seems strange

Reply:
> It should be just object, not function, I didn't mean theme

Ah sorry, I misread that `{|size: string|}` was in the third param for theme, not the second one for props; my bad.

Reply:
> I specified `any` because overload wasn't selecting function without annotation, which is seems strange

Does it infer the type of `props` correctly, though? I'd test it myself, but I am stuck on Windows for the night. 😢

Reply:
Doesn't Flow works on Windows? I also doesn't have PC now, I'm on mobile

Reply:
> Doesn't Flow works on Windows? I also doesn't have PC now, I'm on mobile

Don't have a dev environment setup on Windows. I'll pull this tomorrow, play with it, and let you know.